thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/controllers/AnalysisController.js

```diff
@@ -5,24 +5,56 @@ function AnalysisController(parentController) {
   this.timeSeriesCompareController = new TimeSeriesCompareController(this);
 
   // Event handlers
+  this.analysisView.searchEvent.attach(this.handleSearchEvent.bind(this));
   this.analysisView.applyDataChangeEvent.attach(this.handleApplyAnalysisEvent.bind(this));
 }
 
 AnalysisController.prototype = {
-  handleAppEvent: function () {
-    HASH_SERVICE.refreshWindowHashForRouting('analysis');
+  handleAppEvent() {
+    this.timeSeriesCompareController.destroy();
+    this.analysisView.destroyAnalysisOptions();
     const hashParams = HASH_SERVICE.getParams();
     this.analysisModel.init(hashParams);
     this.analysisModel.update(hashParams);
     this.analysisView.init(hashParams);
-    this.analysisView.render();
-    this.timeSeriesCompareController.handleAppEvent(this.analysisView.viewParams);
+    this.analysisView.render(hashParams.metricId, () => {
+      this.initTimeSeriesController(hashParams);
+    });
   },
 
-  handleApplyAnalysisEvent: function (viewObject) {
-    HASH_SERVICE.update(viewObject.viewParams);
+  handleApplyAnalysisEvent(viewObject) {
+    const params = viewObject.viewParams;
+    this.timeSeriesCompareController.destroy();
+
+    HASH_SERVICE.update(params);
     HASH_SERVICE.refreshWindowHashForRouting('analysis');
-    this.timeSeriesCompareController.handleAppEvent(HASH_SERVICE.getParams());
+    this.initTimeSeriesController(HASH_SERVICE.getParams());
+  },
+
+  handleSearchEvent(params = {}) {
+    const { searchParams } = params;
+    this.timeSeriesCompareController.destroy();
+    this.analysisModel.init();
+    this.analysisView.destroyAnalysisOptions();
+
+    this.analysisModel.fetchAnalysisOptionsData(searchParams.metricId, 'analysis-spin-area').then((res) => {
+      HASH_SERVICE.clear();
+      searchParams.tab = 'analysis';
+      const { currentStart, currentEnd, baselineStart, baselineEnd, granularity } = this.analysisModel;
+      Object.assign(searchParams, {currentStart, currentEnd, baselineStart, baselineEnd, granularity});
+      HASH_SERVICE.update(searchParams);
+      HASH_SERVICE.refreshWindowHashForRouting('analysis');
+      const hashParams = HASH_SERVICE.getParams();
```

Comment:
In this case it wouldn't be possible since hashParams will be updated on line 46. I could possibly declare the variable then set it later on, but it seems to be a bit redundant. Any thought @yee-lkd